modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/client/DefaultExtensionsClient.java

```diff
@@ -0,0 +1,149 @@
+
+  @Inject
+  private MuleContext muleContext;
+
+  private ExtensionManagerAdapter extensionManager;
+
+  private final DefaultPolicyManager policyManager = new DefaultPolicyManager();
+  private final Map<String, ConfigurationProvider> configurations = new LinkedHashMap<>();
+  private final Map<String, OperationModel> operations = new LinkedHashMap<>();
+  private final Map<String, ExtensionModel> extensions = new LinkedHashMap<>();
+
+  /**
+   * {@inheritDoc}
+   * <p>
+   * Set's up the {@link PolicyManager} and the {@link ExtensionManager} used to execute the operations.
+   */
+  @Override
+  public void initialise() throws InitialisationException {
+    this.extensionManager = (ExtensionManagerAdapter) muleContext.getExtensionManager();
+    try {
+      muleContext.getInjector().inject(this.policyManager);
+    } catch (MuleException e) {
+      throw new InitialisationException(e, this.policyManager);
+    }
+    this.policyManager.initialise();
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public <T, A extends Attributes> Publisher<Result<T, A>> executeAsync(String extension, String operation,
+                                                                        OperationParameters parameters)
+      throws MuleException {
+    OperationMessageProcessor processor = createProcessor(extension, operation, parameters);
+    Publisher<Event> publisher = processor.apply(just(getInitialiserEvent(muleContext)));
+    return from(publisher).map(e -> Result.<T, A>builder(e.getMessage()).build());
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public <T, A extends Attributes> Result<T, A> execute(String extension, String operation,
+                                                        OperationParameters parameters)
+      throws MuleException {
+    OperationMessageProcessor processor = createProcessor(extension, operation, parameters);
+    Event process = processor.process(getInitialiserEvent(muleContext));
+    return Result.<T, A>builder(process.getMessage()).build();
+  }
+
+  /**
+   * Creates a new {@link OperationMessageProcessor} for the required operation and parses all the parameters passed by
+   * the client user.
+   */
+  private OperationMessageProcessor createProcessor(String extensionName, String operationName, OperationParameters parameters) {
+    ExtensionModel extension = findExtension(extensionName);
+    OperationModel operation = findOperation(extension, operationName);
+    ConfigurationProvider config = parameters.getConfigName().map(this::findConfiguration).orElse(null);
+    OperationMessageProcessorFactory factory = new OperationMessageProcessorFactory(extension, operation, config, policyManager,
+                                                                                    toResolvers(parameters), muleContext, "");
+    try {
+      OperationMessageProcessor processor = factory.getOperationMessageProcessor();
+      processor.doStart();
+      processor.doInitialise();
```

Comment:
Why do you use the doLifecycle() methods instead of the original ones. 

1) The initialise method already contains logic for locating the ConfigurationProvider. If you could leverage that, it's one less thing to do here
2) The initialise method also validates that the config is not an expression and other things that are worth reusing
3) The start() method makes sure that the doStart() method is executed with the correct classloader, which is necessary. By skipping that, you're missing that condition here and thus getting exposed to trouble.

I think a best effort should be made so that you use the initialise() and start() methods here